scripts/zkai_d64_block_receipt_composition_gate.py

```
@@ -0,0 +1,1010 @@
+        "non_claims": NON_CLAIMS,
+        "validation_commands": VALIDATION_COMMANDS,
+    }
+    refresh_commitments(payload)
+    validate_payload(payload)
+    return payload
+
+
+def _manifest_by_slice(payload: dict[str, Any]) -> dict[str, dict[str, Any]]:
+    manifest = payload.get("source_evidence_manifest")
+    if not isinstance(manifest, list):
+        raise D64BlockReceiptError("source evidence manifest must be a list")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)